src/pyradiosky/skymodel.py

```diff
@@ -4731,163 +4703,12 @@ def read(
-            one column in the file.
-        lat_column : str
-            Part of expected VOTable latitudinal coordinate column. Should match only
-            one column in the file.
-        flux_columns : str or list of str
-            Part of expected vot Flux column(s). Each one should match only one column
-            in the file. Only used for vot files.
-        frame : str
-            Name of coordinate frame for VOTable source positions (lon/lat columns).
-            Defaults to "icrs". Must be interpretable by
-            `astropy.coordinates.frame_transform_graph.lookup_name()`. Only used for
-            vot files.
-        reference_frequency : :class:`astropy.units.Quantity`
-            Reference frequency for VOTable flux values, assumed to be the same value
-            for all components.
-        freq_array : :class:`astropy.units.Quantity`
-            Frequencies corresponding to VOTable flux_columns (should be same length).
-            Required for multiple flux columns.
-        freq_edge_array : :class:`astropy.units.Quantity`
-            Frequency sub-band edges for each flux_columns, shape
-            (2, len(flux_columns)). Required for multiple flux columns if
-            `freq_array` is not regularly spaced. If `freq_array` is regularly
-            spaced and `freq_edge_array` is not passed, `freq_edge_array` will
-            be calculated from the freq_array assuming the band edges are
-            directly between the band centers.
-        spectral_index_column : str
-            Part of expected VOTable spectral index column. Should match only one
-            column in the file.
-        flux_error_columns : str or list of str
-            Part of expected VOTable flux error column(s). Each one should match only
-            one column in the file.
-        history : str
-            History to add to object for VOTable files.
-
-        """
-        self = cls()
-        self.read(
-            filename,
-            filetype=filetype,
-            run_check=run_check,
-            check_extra=check_extra,
-            run_check_acceptability=run_check_acceptability,
-            # Gleam vot
-            spectral_type=spectral_type,
-            with_error=with_error,
-            use_paper_freqs=use_paper_freqs,
-            # fhd
-            expand_extended=expand_extended,
-            # vot
-            table_name=table_name,
-            id_column=id_column,
-            lon_column=lon_column,
-            lat_column=lat_column,
-            flux_columns=flux_columns,
-            frame=frame,
-            reference_frequency=reference_frequency,
-            freq_array=freq_array,
-            freq_edge_array=freq_edge_array,
-            spectral_index_column=spectral_index_column,
-            flux_error_columns=flux_error_columns,
-            history=history,
-        )
-        return self
+    @copy_replace_short_description(read, style=DocstringStyle.NUMPYDOC)
```

Comment:
Are you sure this doesn't need to be `SkyModel.read` instead of just `read`?

Reply:
This is how we do it on pyuvdata and it looks like it works in the built docs.